server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -2182,6 +2182,7 @@ private void removeFullyOverlapFiles(TsFileResource newTsFile, Iterator<TsFileRe
       TsFileResource existingTsFile = iterator.next();
       if (newTsFile.isPlanRangeCovers(existingTsFile)
           && !newTsFile.getTsFile().equals(existingTsFile.getTsFile())
+          && !existingTsFile.isMerging()
```

Comment:
hi ,  I have a qeustion.
if do not delete file which is merging in snapshot.  after merged finished, this tsfile will merge to a new tsfile..
this new tsfile will be overlap and snapshot tsfile..